src/components/TimeSeries/TimeSeriesForm.tsx

```diff
@@ -0,0 +1,90 @@
+// TODO: Fix eslint errors
+
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import { format, subDays } from 'date-fns';
+import { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+import { stations } from '../../mocks/timeseries/stations.json';
+import timeseries from '../../mocks/timeseries/timeseries.json';
+import {
+  type Station,
+  type TimeSeriesResponse
+} from '../../types/models/timeSeries';
+import './TimeSeriesForm.scss';
+
+interface TimeSeriesFormProps {
+  setChartData: (data: TimeSeriesResponse) => void;
+}
+
+const DATE_FORMAT = 'yyyy-MM-dd';
+
+function TimeSeriesForm({ setChartData }: TimeSeriesFormProps) {
+  const [stationsList, setStationsList] = useState<Station[]>([]);
+  const [isError] = useState<boolean>(false);
+
+  const { register, handleSubmit } = useForm({
+    defaultValues: {
+      dateFrom: format(subDays(new Date(), 7), DATE_FORMAT),
+      dateTo: format(new Date(), DATE_FORMAT),
+      types: ['PM2-5'],
+      stations: ['KRK']
+    }
+  });
+
+  const fetchData = async (formData: FormData): Promise<void> => {
```

Comment:
`FormData` is not correct type. You should define your custom type based on data in the form. This type should be the same as the type of data in the form.

Create this type:
```
type TimeSeriesFormData = {
  dateFrom: string;
  dateTo: string;
  types: ('PM2.5' | 'PM10')[],
  stations: string[];
}
```
(If there are more pollution types, let's add them to the 'PM2.5' | 'PM10' union)
I would also prefer to use `PM2.5` notation instead of `PM2-5`.

You can then add this type to the `useForm` hook: `useForm<TimeSeriesFormData>`.

And next, use this type instead of the `FormData` type. in the `fetchData` function.

Reply:
Typing of:
`types: Array<'PM1' | 'PM2.5' | 'PM10'>,`
might be problematic, because I aimed to load types dynamically from the REST endpoint. If we make sure that there won't be any changes in the types of data, we can use union with all types of pollutions. For now, I will leave it as an array of strings, it that's ok.